frontend/src/app/pages/cliente/solicitacao/visualizar-solicitacao/visualizar-solicitacao.component.ts

```diff
@@ -1,36 +1,146 @@
-  templateUrl: './visualizar-solicitacao.component.html',
-  styleUrl: './visualizar-solicitacao.component.css'
-})
-export class VisualizarSolicitacaoComponent implements OnInit {
-    @ViewChild('formSolicitacao') formSolicitacao! : NgForm;
-    solicitacao: Solicitacao = new Solicitacao();
-    id: string = '';
-
-  constructor(
-    private solicitacaoService: SolicitacaoService,
-    private route: ActivatedRoute,
-    private router: Router
-  ) {}
-
-  ngOnInit(): void {
-    let id = +this.route.snapshot.params['id'];
-    const res = this.solicitacaoService.buscarPorId(id);
-    if (res !== undefined) {
-      this.solicitacao = res;
-    } else {
-      throw new Error("Erro ao buscar solicitacao, id = " + id);
-    }
-  }
-
-}
+import { Component, OnInit, ViewChild, TemplateRef } from '@angular/core';
+import { FormsModule, NgForm } from '@angular/forms';
+import { Solicitacao } from '../../../../shared/models/solicitacao.model';
+import { Equipamento } from '../../../../shared/models/equipamento.model';
+import { Orcamento } from '../../../../shared/models/orcamento.model';
+import { ActivatedRoute, RouterModule } from '@angular/router';
+import { SolicitacaoService } from '../../../../services/solicitacao.service';
+import { EquipamentoService } from '../../../../services/equipamento.service';
+import { OrcamentoService } from '../../../../services/orcamento.service';
+import { CommonModule } from '@angular/common';
+import { ModalComponent } from '../../../../components/modal/modal.component';
+import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
+
+@Component({
+  selector: 'app-visualizar-solicitacao',
+  standalone: true,
+  imports: [CommonModule, FormsModule, RouterModule, ModalComponent, NgbModule],
+  templateUrl: './visualizar-solicitacao.component.html',
+  styleUrl: './visualizar-solicitacao.component.css'
+})
+export class VisualizarSolicitacaoComponent implements OnInit {
+
+    @ViewChild('formSolicitacao') formSolicitacao! : NgForm;
+    @ViewChild(ModalComponent) modal!: ModalComponent; 
+    @ViewChild('rejectTemplate') rejectTemplate!: TemplateRef<any>;
+
+    solicitacao: Solicitacao = new Solicitacao();
+    id: number = 0;
+    equipamento?: Equipamento | null;
+    orcamento?: Orcamento | null;
+    isEquipOpen = false;
+    isHistOpen = false;
+
+    currentModalTitle: string = '';
+    currentContentTemplate!: TemplateRef<any>;
+    currentFormData: any = {};
+
+  constructor(
+    private solicitacaoService: SolicitacaoService,
+    private route: ActivatedRoute,
+    private equipamentoService: EquipamentoService,
+    private orcamentoService: OrcamentoService
+  ) {}
+
+  ngOnInit(): void {
+    this.id = +this.route.snapshot.params['id'];
+    const res = this.solicitacaoService.buscarPorId(this.id);
+    if (res !== undefined) {
+      this.solicitacao = res;
+    } else {
+      throw new Error("Erro ao buscar solicitacao, id = " + this.id);
+    }

```

Comment:
Se quiser, isso pode ser trocado para
```suggestion
    if (!res) {
      throw new Error("Erro ao buscar solicitacao, id = " + this.id);
    }
    
    this.solicitacao = res;
    (...)
```